sycl/plugins/unified_runtime/ur/adapters/hip/usm.cpp

```diff
@@ -13,85 +13,86 @@
 #include "context.hpp"
 #include "device.hpp"
 #include "platform.hpp"
+#include "usm.hpp"
 
 /// USM: Implements USM Host allocations using HIP Pinned Memory
-UR_APIEXPORT ur_result_t UR_APICALL urUSMHostAlloc(
-    ur_context_handle_t hContext, const ur_usm_desc_t *pUSMDesc,
-    [[maybe_unused]] ur_usm_pool_handle_t pool, size_t size, void **ppMem) {
-
-  UR_ASSERT(!pUSMDesc || (pUSMDesc->align == 0 ||
-                          ((pUSMDesc->align & (pUSMDesc->align - 1)) == 0)),
+UR_APIEXPORT ur_result_t UR_APICALL
+urUSMHostAlloc(ur_context_handle_t hContext, const ur_usm_desc_t *pUSMDesc,
+               ur_usm_pool_handle_t hPool, size_t size, void **ppMem) {
+  auto alignment = pUSMDesc ? pUSMDesc->align : 0u;
+  UR_ASSERT(!pUSMDesc ||
+                (alignment == 0 || ((alignment & (alignment - 1)) == 0)),
             UR_RESULT_ERROR_INVALID_VALUE);
+  UR_ASSERT(size < hContext->DeviceId->getMaxAllocSize(),
+            UR_RESULT_ERROR_INVALID_USM_SIZE);
```

Comment:
Would it be worth to fold the alignment and max alloc checks into a helper, all 3 alloc funcs do exact the same work.

Reply:
We shouldn't have the max alloc checks, see:
* https://github.com/intel/llvm/pull/11023

The 1/4 of global memory is a leftover from OpenCL and not accurate to what the hardware support so adding this check will create regressions in benchmarks that use larger allocations.

In addition HIP will return an error anyway if it can't handle the allocation, so we don't need the extra check.

Reply:
I've created a helper function for the alignment check and removed the max alloc check.